File: src/main/kotlin/riesgo/controlador/Game.java
package riesgo.controlador;

import riesgo.Colors;
import riesgo.view.Jugador1;

import java.io.FileDescriptor;
import java.io.FileOutputStream;
import java.io.PrintStream;
import java.nio.charset.StandardCharsets;
import java.util.Scanner;

import static riesgo.view.MainKt.Jugador1;

class Game {

public void joc() {
        // Set the charset to UTF 8 to the Standard Output
        if (!StandardCharsets.UTF_8.name().equals(System.out.getClass().getName())) {
        System.setOut(new PrintStream(new FileOutputStream(FileDescriptor.out), true, StandardCharsets.UTF_8));
        }
        boolean fideljoc = true;

        //FEM UN BUCLE WHILE QUE SERÀ L'EXECUCIÓ DEL JOC EN SI.
        while (fideljoc) {
        System.out.print("\u001b[2J");

        //PRESENTEM EL JOC D'UNA MANERA AMENA I AGRADABLE A LA VISTA
        System.out.println("\n==========================================================================================");
        System.out.println("=============BENVINGUT A \"RIESGO\", EL RISK IBÈRIC=========================================");
        System.out.println("==========================================================================================");
        System.out.println("=============MENÚ DEL JOC=================================================================");
        System.out.println("==========================================================================================");


        //DEMANEM LES CREDENCIALS ALS JUGADORS. (NOM I COLOR)
        System.out.println("Nom del jugador 1: ");
        Scanner scanner = new Scanner(System.in);
        Jugador1.nom = (scanner.hasNextLine()) ? scanner.nextLine() : "Unnamed1";
        System.out.println("Color del jugador 1: ");
        System.out.println("LLEGENDA DE COLORS:");
        System.out.println("1-vermell, 2-verde que te quiero verde, 3-blau, 4-groc, 5-rosa que l'amor s'hi posa");
        try {
        switch (scanner.nextInt()) {
        case 1:
        Jugador1.color = Colors.ANSI_RED;
        break;
        case 2:
        Jugador1.color = Colors.ANSI_GREEN;
        break;
        case 3:
        Jugador1.color = Colors.ANSI_BLUE;
        break;
        case 4:
        Jugador1.color = Colors.ANSI_YELLOW;
        break;
        case 5:
        Jugador1.color = Colors.ANSI_PURPLE;
        break;
default:
        Jugador1.color = Colors.ANSI_RED;
        break;
        }
        } catch (Exception e) {
        Jugador1.color = Colors.ANSI_RED;
        }
        Jugador1.numSoldats = 30;


        System

File: src/main/kotlin/riesgo/controlador/Game.kt
package riesgo.controlador

import riesgo.Colors
import riesgo.view.Jugador1
import riesgo.view.Jugador2
import java.io.FileDescriptor
import java.io.FileOutputStream
import java.io.PrintStream
import java.nio.charset.StandardCharsets

class Game {

    fun joc() {

        // Set the charset to UTF 8 to the Standard Output
        if (System.out.charset() != StandardCharsets.UTF_8) {
            System.setOut(PrintStream(FileOutputStream(FileDescriptor.out), true, StandardCharsets.UTF_8))
        }
        val fideljoc = true

        //FEM UN BUCLE WHILE QUE SERÀ L'EXECUCIÓ DEL JOC EN SI.
        while (fideljoc) {
            print("\u001b[2J")

            //PRESENTEM EL JOC D'UNA MANERA AMENA I AGRADABLE A LA VISTA
            println("\n==========================================================================================")
            println("=============BENVINGUT A \"RIESGO\", EL RISK IBÈRIC=========================================")
            println("==========================================================================================")
            println("=============MENÚ DEL JOC=================================================================")
            println("==========================================================================================")


            //DEMANEM LES CREDENCIALS ALS JUGADORS. (NOM I COLOR)
            println("Nom del jugador 1: ")
            Jugador1.nom = readlnOrNull() ?: "Unnamed1"
            println("Color del jugador 1: ")
            println("LLEGENDA DE COLORS:")
            println("1-vermell, 2-verde que te quiero verde, 3-blau, 4-groc, 5-rosa que l'amor s'hi posa")
            when (readln().toIntOrNull()) {
                1 -> {
                    Jugador1.color = Colors.ANSI_RED
                }

                2 -> {
                    Jugador1.color = Colors.ANSI_GREEN
                }

                3 -> {
                    Jugador1.color = Colors.ANSI_BLUE
                }

                4 -> {
                    Jugador1.color = Colors.ANSI_YELLOW
                }

                5 -> {
                    Jugador1.color = Colors.ANSI_PURPLE
                }

                else -> {
                    Jugador1.color = Colors.ANSI_RED
                }
            }
            Jugador1.numSoldats = 30


            println("Nom del jugador 2: ")
            Jugador2.nom = readlnOrNull() ?: "Unnamed2"
            println("Color del jugador 2: ")
            println("LLEGENDA DE COLORS:")
            println("1-vermell, 2-verde que te quiero verde, 3-blau, 4-groc, 5-rosa que l'amor s'hi posa")
            when (readln().toIntOrNull()) {
                1 -> {
                    Jugador2.color = Colors.ANSI_RED
                }

                2 -> {
                    Jugador2.color = Colors.ANSI_GREEN
                }

                3 -> {
                    Jugador2.color = Colors.ANSI_BLUE
                }

                4 -> {
                    Jugador2.color = Colors.ANSI_YELLOW
                }

                5 -> {
                    Jugador2.color = Colors.ANSI_PURPLE
                }

                else -> {
                    Jugador2.color = Colors.ANSI_GREEN
                }
            }
            Jugador2.numSoldats = 30


            print("\u001b[2J")

            print("\n")
            val torns = TurnsFunctions()
            torns.tornsDeTot()
        }
    }
}